### PR TITLE
Add ability to disable staking validation for requester engine

### PR DESCRIPTION
**Pull request by m4ksio**
See #5727 for explanation

### CODE REVIEW

File: engine/common/requester/engine_test.go [hunk 1]
```diff
@@ -391,3 +394,90 @@ func TestOnEntityIntegrityCheck(t *testing.T) {
 	// make sure we process item without checking integrity
 	assert.Equal(t, 1, called)
 }
+
+func TestOriginValidation(t *testing.T) {
```
Comment (zhangchiqing) [inline engine/common/requester/engine_test.go:398]:
```suggestion
// Verify that the origin should not be checked when ValidateStaking config is set to false
func TestOriginValidation(t *testing.T) {
```

**Review by zhangchiqing — APPROVED** [review]

File: cmd/execution/main.go [hunk 2]
```diff
@@ -308,10 +308,13 @@ func main() {
 		Component("ingestion engine", func(builder cmd.NodeBuilder, node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
 			collectionRequester, err = requester.New(node.Logger, node.Metrics.Engine, node.Network, node.Me, node.State,
 				engine.RequestCollections,
-				filter.HasRole(flow.RoleCollection),
+				filter.Any,
```
Comment (ramtinms) [inline cmd/execution/main.go:311]: I assume we do collection hash somewhere that we let this open to any

Comment (m4ksio) [reply cmd/execution/main.go:311]: Yep - https://github.com/onflow/flow-go/blob/36069d4273be79fe41268c6abac8a4eb150412a0/engine/common/requester/engine.go#L447

File: engine/common/requester/engine.go [hunk 3]
```diff
@@ -80,14 +81,20 @@ func New(log zerolog.Logger, metrics module.EngineMetrics, net module.Network, m
...
-		filter.Not(filter.Ejected),
 		filter.Not(filter.HasNodeID(me.NodeID())),
 	)
 
+	// make sure we don't send requests to unstaked nodes
+	if cfg.ValidateStaking {
+		selector = filter.And(
+			selector,
+			filter.HasStake(true),
+			filter.Not(filter.Ejected))
```
Comment (huitseeker) [inline engine/common/requester/engine.go:95]: I like the idea of ejection being a mechanism that works irrespective of whether you're staked or not.

**Review by huitseeker — COMMENTED** [review]
Apart from my one comment (I'd like to have ejection work even for unstaked nodes), LGTM

**Review by huitseeker — APPROVED** [review]

File: engine/common/requester/engine.go (hunk 3 above)
Comment (m4ksio) [reply engine/common/requester/engine.go:95]: I'm not sure what you mean...

Comment (huitseeker) [reply engine/common/requester/engine.go:95]: I would apply `filter.Not(filter.Ejected)` outside of the conditional `if cfg.ValidateStaking`

**Comment by huitseeker** [comment]
this could be a great PR to try bors with btw

**Review by ramtinms — APPROVED** [review]

File: engine/common/requester/engine.go (hunk 3 above)
Comment (m4ksio) [reply engine/common/requester/engine.go:95]: Ah right, moved so its applies always

**Pull request merged**